src/Cli/dotnet/Commands/Test/Terminal/TerminalTestReporter.cs

```diff
@@ -479,12 +466,34 @@ internal void TestCompleted(
                 break;
         }
 
+        if (asm.PreviousTestInstanceIds.TryGetValue(testNodeUid, out var previousRuns))
+        {
+            // Different InstanceId means that we are running in a new process instance but in the
+            // same executionId, so the test is being retried via the Retry extension.
+            if (previousRuns.Count > 0 && !previousRuns.Contains(instanceId))
+            {
+                // This is a retried test, fix the counts.
+                // Previous test failed because we are retrying.
+                // We don't need to be atomic, because the update is done below
+                // where we update the worker to make it re-render.
+                asm.FailedTests--;
```

Comment:
Does this fix the count correctly?

What if there we get different number of test cases of the same TestNodeUid in each run? This is the case with at least MSTest's DynamicData when it's folded.

Reply:
It did not, as we discussed I moved to a different approach.